home-assignments/session1/solution1.py

```diff
@@ -0,0 +1,77 @@
+import json
+import yaml
+import argparse
+
+
+# TODO: fix encoding issue
+
+
+def import_data_from_file(file_name):
+    j_file = file_name
+    try:
+        with open(j_file, encoding="utf-8") as jfile:
+            output = json.load(jfile)
+
+            return output
+    except Exception as e:
+        print(e)
+        exit(1)
+
+
+def analyze_input_data(**output):
+    ppl_ages = output['ppl_ages']
+    buckets = output['buckets']
+    all_ages = []
+    for name, age in ppl_ages.items():
+        all_ages.append(age)
+
+    max_age = max(all_ages)
+    max_partition = max(buckets)
+    if max_age > max_partition:
+        buckets.append(max_age)
+
+    buckets.append(0)
+    buckets = sorted(buckets)
+    results = (ppl_ages, buckets)
```

Comment:
You don't have to use "results" parameter. You can just do: return (ppl_ages, buckets)

Reply:
i will leave it that way, as it is same parameter that used  as input for next function.